lib/auth/clt.go

```diff
@@ -96,22 +96,19 @@ func NewClient(cfg client.Config, params ...roundtrip.ClientParam) (*Client, err
 		return nil, trace.Wrap(err)
 	}
 
-	// This logic is necessary for the client to force client to always send
-	// a certificate regardless of the server setting. Otherwise the client may pick
-	// not to send the client certificate by looking at certificate request.
-	if len(cfg.TLS.Certificates) != 0 {
-		cert := cfg.TLS.Certificates[0]
-		cfg.TLS.Certificates = nil
-		cfg.TLS.GetClientCertificate = func(_ *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return &cert, nil
-		}
+	// Many uses of the lib/client do not expect an open valid connection immediately after
+	// initialization, so NoPingCheck is used for backwards compatibility with this client.
+	cfg.NoPingCheck = true
```

Comment:
I asked above too, but what does "open valid connection" mean exactly? If the server isn't listening, grpc.Dial will fail too, no?